### ISSUE TITLE
Logitech F310 discussion

### ISSUE THREAD

**Issue by Narann** [mupen64plus/mupen64plus-input-sdl#37]
[This commit](https://github.com/mupen64plus/mupen64plus-input-sdl/commit/bcf6892ca613ae6471ddb01c15792878f30bad67) break my previously working Logitech F310. Buttons are not correctly mapped while the original Xbox layout worked fine.

Looking in the google group, I paste the mail which originate the change here:

> To whom it may concern:
> 
> I recently purchased a Logitech Gamepad F310 to use with Mupen64Plus and
> found that the Z button was mapped to the Start button and a few other
> oddities.  When I looked in the InputAutoCfg.ini there was nothing under
> the sub-heading for this particular controller, it was just listed under
> "XBox Clones" with no configurations present for it.  Here's a
> configuration I created to place a few buttons where they're supposed to
> be.  I put N64's "B" on the "X" (right) face button because it felt more
> natural to me, N64's "Z" is set to RB, and the LT and RT buttons are
> mapped as the N64's L/R buttons.  The rest of the config is pretty much
> identical to other Logitech Playstation-esque controllers.

Here is what I suspect: Logitech F310 is a XBox mapped controler having a legacy "directinput" mode in the back:

![f310-gaming-gamepad 1](https://cloud.githubusercontent.com/assets/659708/20132784/de1804b2-a664-11e6-8d93-2d18ad72b4a2.jpg)

I still have to check but I suspect the XBox layout break in direct input mode. Maybe it's [related to OS](http://wiki.unity3d.com/index.php?title=Xbox360Controller#Axises).

Having a different, dedicated mapping for a controller mapping a Xbox layout is kinda weird. It would be interesting to ask for the OS when users provide a mapping for an already existing gamepad (mine is Linux Mint 17.1).

**Comment by richard42** [comment]
Yes, the button mapping changes depending upon whether this switch is set to D or X.  I'm not sure whether we can accommodate both mappings with a single config file.

**Comment by Narann** [comment]
TBH, as it's a simple switch, I don't see any reason to support in DirectInput mode. Xbox made something cool providing a bunch of compatible controllers.

What I suggest is to put some word about it on our wiki.

Once again, I still have to test. :)
